File: eMovieFinder/eMovieFinder.Model/SearchObjects/MovieReviewSearchObject.cs
namespace eMovieFinder.Model.SearchObjects
{
    public class MovieReviewSearchObject : BaseSearchObject
    {
        public int? UserId { get; set; }
        public int? MovieId { get; set; }
        public bool? IsUserIncluded { get; set; }
        public bool? IsMovieIncluded { get; set; }
    }
}

File: eMovieFinder-UI/emovie_finder_mobile/lib/src/models/searchobjects/movie_review_search_object.dart
import 'package:json_annotation/json_annotation.dart';

import 'base_search_object.dart';

part 'movie_review_search_object.g.dart';

@JsonSerializable()
class MovieReviewSearchObject extends BaseSearchObject {
  final int? userId;
  final int? movieId;
  final bool? isUserIncluded;
  final bool? isMovieIncluded;

  MovieReviewSearchObject({
    super.pageNumber,
    super.pageSize,
    super.orderBy,
    super.isDescending,
    this.userId,
    this.movieId,
    this.isUserIncluded,
    this.isMovieIncluded,
  });

  factory MovieReviewSearchObject.fromJson(Map<String, dynamic> json) => _$MovieReviewSearchObjectFromJson(json);
  Map<String, dynamic> toJson() => _$MovieReviewSearchObjectToJson(this);
}